### PR TITLE
deltaDebug.py  & removeInst() 

### PR DESCRIPTION
**Pull request by Mahfouz-z**
Adding the new script for delta debugging under etc.
Improving _dbModule::removeInst() through the use of a doubly linked list.

### CODE REVIEW

File: src/odb/src/db/dbModule.cpp [hunk 1]
```diff
@@ -212,36 +212,56 @@ void dbModule::addInst(dbInst* inst)
   }
 
   _inst->_module = module->getOID();
-  _inst->_module_next = module->_insts;
+
+  if (module->_insts != 0) {
+    _dbInst* _tail = block->_inst_tbl->getPtr(module->_insts);
```
Comment (maliberty) [inline src/odb/src/db/dbModule.cpp:217]: Isn't this the current head?  Local variables should not have a lead '_'

Comment (Mahfouz-z) [reply src/odb/src/db/dbModule.cpp:217]: yes, it is; will fix it.

**Review by maliberty — CHANGES_REQUESTED** [review]

File: etc/deltaDebug.py [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+        self.use_stdout = use_stdout
+        self.step_count = 0
+
+        # timeout used to measure the time the original input takes
+        # to reach an error to use as standard timeout for different 
+        # cuts.
+        self.timeout = 1e6 # Timeout in seconds
+        
+        # Setting persistence for the run
+        if opt.persistence in range(1,7):
```
Comment (maliberty) [inline etc/deltaDebug.py:60]: You can enforce this with choices in argparse

File: etc/deltaDebug.py [hunk 3]
```diff
@@ -0,0 +1,304 @@
...
+
+
+parser = argparse.ArgumentParser('Arguments for delta debugging')
+parser.add_argument('--base_db_path', type=str, help='Path to the db file to perform the step on')
+parser.add_argument('--error_string', type=str, help='The output that indicates target error has occured')
+parser.add_argument('--step', type=str, help='Command used to perform step on the input odb file')
+parser.add_argument('--persistence', type=int, default=1, choices=[1,2,3,4,5,6], help= 'Indicates maximum input fragmentation; fragments = 2^persistence; value in [1,6]')
+
+dump_def = 0    # a flag to indicate whether to dump def per each step, could be used if the step acts on a def file rather than an odb
+use_stdout = 0  # a flag either 0 or 1 to enable/disable detecting the error string from stdout in addition to stderr
```
Comment (maliberty) [inline etc/deltaDebug.py:40]: Why aren't these command line args anymore?  I mean you should use 

parser.add_argument('--use_stdout', action='store_true')

Comment (Mahfouz-z) [reply etc/deltaDebug.py]: Sorry, miss understood what you meant by this edit. Fixed.

**Review by maliberty — APPROVED** [review]

**Pull request merged**